knowledge/knowledge_read.go

```diff
@@ -23,190 +23,188 @@ import (
 	ctx context.Context,
 	identifier *Identifier,
 	opts ...grpc.CallOption,
-) (*knowledgepb.Node, error) {
-	return c.getNodeByIdentifier(ctx, DigitalTwin, identifier, opts...)
+) (*knowledgeobjects.Node, error) {
+	return c.GetNodeByIdentifier(ctx, identifier, true, opts...)
 }
 
-// GetResourceByID is a helper function that queries for a Resource node by its id.
-func (c *Client) GetResourceByID(
-	ctx context.Context,
-	id string,
-	opts ...grpc.CallOption,
-) (*knowledgepb.Node, error) {
-	ctx = insertMetadata(ctx, c.xMetadata)
-	return c.getNodeByID(ctx, id, Resource, opts...)
-}
-
-// GetResourceByIdentifier is a helper function that queries for a Resource node
-// by its externalID + type combination.
-func (c *Client) GetResourceByIdentifier(
-	ctx context.Context,
-	identifier *Identifier,
-	opts ...grpc.CallOption,
-) (*knowledgepb.Node, error) {
-	ctx = insertMetadata(ctx, c.xMetadata)
-	return c.getNodeByIdentifier(ctx, Resource, identifier, opts...)
-}
-
-// ListDigitalTwinsByProperty is a helper function that lists all DigitalTwin nodes
+// ListIdentitiesByProperty is a helper function that lists all Identity nodes
 // that have the specified property.
-func (c *Client) ListDigitalTwinsByProperty(
-	ctx context.Context,
-	property *knowledgepb.Property,
-	opts ...grpc.CallOption) ([]*knowledgepb.Node, error) {
-	return c.ListNodesByProperty(ctx, DigitalTwin, property, opts...)
-}
-
-// ListResourcesByProperty is a helper function that lists all Resource nodes.
-// that have the specified property.
-func (c *Client) ListResourcesByProperty(
-	ctx context.Context,
-	property *knowledgepb.Property,
-	opts ...grpc.CallOption) ([]*knowledgepb.Node, error) {
-	return c.ListNodesByProperty(ctx, Resource, property, opts...)
-}
-
-// ListDigitalTwins is a helper function that lists all DigitalTwin nodes.
-func (c *Client) ListDigitalTwins(
+func (c *Client) ListIdentitiesByProperty(
 	ctx context.Context,
-	opts ...grpc.CallOption) ([]*knowledgepb.Node, error) {
-	return c.ListNodes(ctx, DigitalTwin, opts...)
+	property *knowledgeobjects.Property,
+	opts ...grpc.CallOption) ([]*knowledgeobjects.Node, error) {
+	return c.ListNodesByProperty(ctx, property, true, opts...)
 }
 
-// ListResources is a helper function that lists all Resource nodes.
-// that have the specified property.
-func (c *Client) ListResources(
+// ListIdentities is a helper function that lists all Identity nodes.
+func (c *Client) ListIdentities(
 	ctx context.Context,
-	opts ...grpc.CallOption) ([]*knowledgepb.Node, error) {
-	return c.ListNodes(ctx, Resource, opts...)
+	opts ...grpc.CallOption) ([]*knowledgeobjects.Node, error) {
+	return c.ListNodes(ctx, true, opts...)
 }
 
-// ListNodes is a helper function that lists all nodes by type, regardless of whether they are DigitalTwins
-// or Resources. The nodeType argument should be in PascalCase.
+// ListNodes is a helper function that lists all nodes by type, regardless of whether they are Identities
+// or not.
```

Comment:
The functionality does not correspond with the comment. Why did you change the logic? Now the function lists all nodes that are either identities or not identities, depending on the arguments. Before, it listed all nodes of the provided nodeType argument, regardless of whether they were identities or not.

Reply:
updated